src/tests/operations/query/r4/r4.test.js

```diff
@@ -892,64 +892,103 @@ describe('r4 search Tests', () => {
                         {
-                            $or: [
-                                {
-                                    effectiveDateTime: {
-                                        $options: 'i',
-                                        $regex: /^(?:2019-10-16T22:12)|(?:2019-10-16T22:12:29)|(?:2019$)|(?:2019-10$)|(?:2019-10-16$)|(?:2019-10-16T22:12Z?$)/
-                                    }
-                                },
-                                {
-                                    $and: [
-                                        {
-                                            'effectivePeriod.start': {
-                                                $lte: '2019-10-16T22:12:29+00:00'
-                                            }
-                                        },
-                                        {
-                                            $or: [
-                                                {
-                                                    'effectivePeriod.end': {
-                                                        $gte: '2019-10-16T22:12:29+00:00'
-                                                    }
-                                                },
-                                                {
-                                                    'effectivePeriod.end': null
-                                                }
-                                            ]
-                                        }
-                                    ]
-                                },
+                          'effectivePeriod.start': {
+                            $lte: '2019-10-16T22:12:29+00:00',
+                            $type: 'string'
+                          }
+                        },
+                        {
+                          $or: [
+                            {
+                              $and: [
                                 {
-                                    effectiveTiming: {
-                                        $options: 'i',
-                                        $regex: /^(?:2019-10-16T22:12)|(?:2019-10-16T22:12:29)|(?:2019$)|(?:2019-10$)|(?:2019-10-16$)|(?:2019-10-16T22:12Z?$)/
-                                    }
+                                  'effectivePeriod.end': {
+                                    $gte: '2019-10-16T22:12:29+00:00'
+                                  }
                                 },
                                 {
-                                    effectiveInstant: {
-                                        $options: 'i',
-                                        $regex: /^(?:2019-10-16T22:12)|(?:2019-10-16T22:12:29)|(?:2019$)|(?:2019-10$)|(?:2019-10-16$)|(?:2019-10-16T22:12Z?$)/
-                                    }
+                                  'effectivePeriod.end': {
+                                    $type: 'string'
+                                  }
```

Comment:
Can we combine $type and $gte, $lte checks? Just like we did here https://github.com/icanbwell/fhir-server/pull/1635/files#diff-c011e3d57fc12bb2b1b701c9958c16f25e649228eff2c05eff23967fa6a261b1R964-R970

Also, we should not change the queries for fields that were already date-type as these fields worked as expected. Like meta.lastUpdated in all resources and recorded field AuditEvent.

Reply:
Updated code to combine checks. Date-type queries only changed to add $type: 'date', to match the 'string' logic.

Reply:
```query before: {'recorded':{'$gt':ISODate('2021-04-02T00:00:00.000Z'),'$lt':ISODate('2021-09-02T00:00:00.000Z')}}
query after: {'$and':[{'recorded':{'$gt':ISODate('2021-04-02T00:00:00.000Z'),'$type':'date'}},{'recorded':{'$lt':ISODate('2021-09-02T00:00:00.000Z'),'$type':'date'}}]}```

We can see that $type: 'date' is added but $lt and $gt checks are separated into 2 queries. Can we avoid this?

Reply:
Updated code to combine $lt and $gt checks for this